pkg/workloads/deployment.go

```diff
@@ -101,3 +102,14 @@ func RequestDeployment(request DeploymentRequest) (*appsv1.Deployment, error) {
 
 	return deployment, nil
 }
+
+func UpdateNodePlacement(existing *appsv1.Deployment, deploy *appsv1.Deployment, changed *bool) {
+	if !reflect.DeepEqual(existing.Spec.Template.Spec.NodeSelector, deploy.Spec.Template.Spec.NodeSelector) {
+		existing.Spec.Template.Spec.NodeSelector = deploy.Spec.Template.Spec.NodeSelector
+		*changed = true
+	}
+	if !reflect.DeepEqual(existing.Spec.Template.Spec.Tolerations, deploy.Spec.Template.Spec.Tolerations) {
+		existing.Spec.Template.Spec.Tolerations = deploy.Spec.Template.Spec.Tolerations
+		*changed = true
+	}
+}
```

Comment:
I don't think this function belongs here 
seems like this is something that could just be handled by `UpdateIfChanged` and we may not need this function at all

Reply:
Good catch.